zerver/tests/test_decorators.py

```diff
@@ -759,6 +759,30 @@ def test_login_deactivated_user(self):
             "Sorry for the trouble, but your account has been deactivated",
             result)
 
+    def test_login_deactivated_mirror_dummy(self):
+        # type: () -> None
+        """
+        logging in fails with an inactive user
+
+        """
+        from zerver.forms import OurAuthenticationForm
+        from zerver.lib.test_classes import initial_password
+
+        user_profile = self.example_user('hamlet')
+        do_deactivate_user(user_profile)
+        user_profile.is_mirror_dummy = True
+        user_profile.save()
+
+        password = initial_password(user_profile.email)
+        request = mock.MagicMock()
+        request.get_host.return_value = 'testserver'
+        form = OurAuthenticationForm(request,
+                                     data={'username': user_profile.email,
+                                           'password': password})
+        with self.settings(AUTHENTICATION_BACKENDS=('zproject.backends.EmailAuthBackend',)):
+            print(form.is_valid())
+            print(form.errors)
```

Comment:
You have a static analysis error related to these print functions not having the right import.  But is print what you intend here? Generally we don't print things in tests.